### ISSUE TITLE
Add ability to reorder files in a package

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#932]
In the editor, add up/down arrows and/or drag-n-drop ability to each row in the table. Reordering the files will reorder the entity sections in the EML. This affects the order of display in the MetadataView.

**Comment by jagoldstein** [comment]
this is definitely something that `submitter`s have asked for repeatedly

**Comment by laurenwalker** [comment]
I'll bump this up to the next minor release

**Comment by laurenwalker** [comment]
Because of the portals work being put on fast track, I'm going to punt this feature

**Comment by laijasmine** [comment]
Bumping this issue as there has been a resurgence in interest on this feature on the ADC. Mainly submitters want to be able to sort by file name in some way (eg. alphabetically) for datasets with 50 + files. Drag and drop or up and down arrows might not work as well in this case.

**Comment by laurenwalker** [comment]
This should be somewhat straightforward. We could have an option to sort the files in the table alphabetically, but then users can drag-n-drop individual files to manually reorder them too.

**Comment by laijasmine** [comment]
that would be great!

**Comment by justinkadi** [comment]
This feature has been a more common request recently from submitters, and I think would be a very useful feature. 

Previously, we used to be able to re-organize the files in a data package by changing the order of the files in the EML. 

Recently, after some backend changes, we've found that the order of files is dependent on the ordering of the files in the resource map file with the `atLocation` element. I've attempted to change the ordering of the `atLocation` elements to change the ordering of the files displayed in the data package. But, when I use the `updateObject` function from the [dataONE R library](https://www.rdocumentation.org/packages/dataone/versions/2.2.2/topics/updateObject) to update the resource map of a data package, it automatically re-orders it again. I haven't figured out how it re-orders it.

Example Tickets requesting feature:
[Shruti Khana Ticket](https://support.nceas.ucsb.edu/rt/Ticket/Display.html?id=28897)
[Dana Brown Ticket](https://support.nceas.ucsb.edu/rt/Ticket/Display.html?id=31136)
[Baptiste Vandecrux Ticket](https://support.nceas.ucsb.edu/rt/Ticket/Display.html?id=31481)

but there have been more requests before as well.

**Comment by mbjones** [comment]
@justinkadi the R package uses the `redland` R package to handle the resource map serialization to RDF. As I've mentioned, there is no innate order to RDF triples, and serializations are free to reorder triples as they see fit. One test you might try is to manually edit the resource map order in RDF/XML format (not using the R client) and then upload that document with curl. I think you will see the RM ordering stays the same when you access it from the server, but the client tools in R are what are changing the ordering around.

**Comment by justinkadi** [comment]
Right now, it's the order of the `aggregates` element in the rdf that determines the order.

**Comment by mbjones** [comment]
Thanks for checking that. Totally erroneous behavior and we shouldn't rely on it. RDF serializations have no inherent order. That MetacatUI uses that haphazard order could be considered a bug.

**Comment by robyngit** [comment]
> Previously, we used to be able to re-organize the files in a data package by changing the order of the files in the EML. Recently, after some backend changes, we've found that the order of files is dependent on the ordering of the files in the resource map file with the atLocation element.

This may be related to the changes we made to `DataPackageView` to display the folder structure from atLocation data.

The issue with getting the order of files from the EML is that we don't parse the EML on the data set landing page. I suppose we could inspect the HTML returned from the view service to get the file order. Is there any other source of information on the preferred order of files?

@justinkadi how do users typically want files ordered? Jasmine mentioned alphabetically in 2021, is that still the case?

**Comment by mbjones** [comment]
Yeah, I don't think its reasonable to have a fixed, arbitrary order to the files. Once they are in folders, they would need to be sorted and paged too. can't we just sort alphabetically or by modification date like most file listings do? If you're on your Mac, what is the "natural" order of files in a folder? I don't think there is one, and you can only sort by file attributes within folders.

**Comment by justinkadi** [comment]
Submitters have often asked to order their files alphabetically. If their file names are organized with dates/numbers, then they often want it listed in that order. Another common request is that they want the README file they include to be listed at the top.

In my Mac, it seems like the default is Folders first, then numerically, then alphabetically. But in my Downloads folder specifically, it's by most recent download ("Date Added")

I just recently [re-ordered files in a data package](https://arcticdata.io/catalog/view/urn%3Auuid%3A73367272-3143-4605-97f2-c94a3e01fb48) that also had folder hierarchy, and that was interesting because the order of the files in the rdf were mixed across folders, but the files still were listed in the correct folder.

**Comment by mbjones** [comment]
https://github.com/user-attachments/assets/a6ab19c3-21a6-47f4-b131-9d3d84b3e553

**Comment by justinkadi** [comment]
Turns out, `aggregates` doesn't strictly define the order either in the resource map. I tried re-ordering them in this [data package](https://doi.org/10.18739/A2FQ9Q753), but it is still in the wrong order.

I also checked `isDocumentedBy`, `identifier`, `isAggregatedBy`, and `documents`.
